algebird-test/src/test/scala/com/twitter/algebird/MomentsLaws.scala

```diff
@@ -1,24 +1,48 @@
 package com.twitter.algebird
 
 import com.twitter.algebird.BaseProperties._
-import com.twitter.algebird.scalacheck.arbitrary._
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
+import org.scalacheck.{Arbitrary, Gen}
 
 class MomentsLaws extends CheckProperties {
   val EPS = 1e-10
 
+  implicit val equiv: Equiv[Moments] =
+    Equiv.fromFunction { (ml, mr) =>
+      (ml.m0 == mr.m0) &&
+      approxEq(EPS)(ml.m1, mr.m1) &&
+      approxEq(EPS)(ml.m2, mr.m2) &&
+      approxEq(EPS)(ml.m3, mr.m3) &&
+      approxEq(EPS)(ml.m4, mr.m4)
+    }
+
+  def opBasedGen[A: Numeric](genA: Gen[A]): Gen[Moments] = {
+    val init: Gen[Moments] = genA.map(Moments(_))
+
+    val recur = Gen.lzy(opBasedGen[A](genA))
+    val pair = Gen.zip(recur, recur)
+
+    import Operators.Ops
+
+    Gen.frequency(
+      (10, init),
+      (1, pair.map { case (a, b) => a + b })
```

Comment:
this is testing the API as the user would use it. It avoids making highly unusual Moments which might hide laws passing (as happened with Group tests previously).

There are really only two things you can do: create a Moments from a value, and combine it. This is generating both kinds.